### PR TITLE
Add key to on change updates

### PR DESCRIPTION
**Pull request by zbud-msft**
#### Why I did it

Key is missing in on change updates

Current:

root@efb973f6ce02:~# python /root/gnxi/gnmi_cli_py/py_gnmicli.py -g -t X -p 50051 -m subscribe -x NEIGH_STATE_TABLE -xt STATE_DB -o ndastreamingservertest --subscribe_mode 0 --submode 1 --interval 0 --update_count 2 --create_connections 1

```
2023-07-20 22:05:25.146156 response received: 
sync_response: true

2023-07-20 22:05:31.271959 response received: 
update {
  timestamp: 1689890731162239986
  prefix {
    target: "STATE_DB"
  }
  update {
    path {
      elem {
        name: "NEIGH_STATE_TABLE"
      }
    }
    val {
      json_ietf_val: "{\"peerType\":\"e-BGP\",\"state\":\"Active\"}"
    }
  }
}
```

Changed:

```
2023-07-20 22:01:56.251096 response received: 
sync_response: true

2023-07-20 22:02:15.558270 response received: 
update {
  timestamp: 1689890542962488225
  prefix {
    target: "STATE_DB"
  }
  update {
    path {
      elem {
        name: "NEIGH_STATE_TABLE"
      }
    }
    val {
      json_ietf_val: "{\"10.0.0.57\":{\"peerType\":\"e-BGP\",\"state\":\"Active\"}}"
    }
  }
}
```
#### How I did it

Set use key value to true

#### How to verify it

E2E test (sonic-mgmt test case)

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Comment by zbud-msft** [comment]
Will add sonic-mgmt E2E test for this case

**Comment by zbud-msft** [comment]
> Will add sonic-mgmt E2E test for this case

https://github.com/sonic-net/sonic-mgmt/pull/9073

File: sonic_data_client/db_client.go [hunk 1]
```diff
@@ -1089,7 +1089,7 @@ func dbSingleTableKeySubscribe(c *DbClient, rsd redisSubData, updateChannel chan
 						continue
 					}
 					tblPath.tableKey = subscr.Channel[prefixLen:]
-					err = tableData2Msi(&tblPath, false, nil, &newMsi)
+					err = tableData2Msi(&tblPath, true, nil, &newMsi)
```
Comment (qiluo-msft) [inline sonic_data_client/db_client.go:1092]: >true [](http://example.com/codeflow?start=35&length=4)

Seems this is the first time `true` is used for this function. Let's add extensive unit test?

**Comment by qiluo-msft** [comment]
Could you reformat the PR description? Some well intentation is lost due to markdown rendering.

**Comment by zbud-msft** [comment]
> Could you reformat the PR description? Some well intentation is lost due to markdown rendering.

Done

File: sonic_data_client/db_client.go (hunk 1 above)
Comment (qiluo-msft) [reply sonic_data_client/db_client.go:1092]: To be more clear, could you add unit test to `tableData2Msi` to cover the second argument as `true`?

Comment (zbud-msft) [reply sonic_data_client/db_client.go:1123]: Added UT

File: sonic_data_client/db_client.go [hunk 2]
```diff
@@ -79,9 +79,20 @@ var IntervalTicker = func(interval time.Duration) <-chan time.Time {
 	return time.After(interval)
 }
 
+var ExportTableData2Msi = tableData2Msi
```
Comment (qiluo-msft) [inline sonic_data_client/db_client.go:82]: >tableData2Msi [](http://example.com/codeflow?start=26&length=13)

Is it better just to change to `TableData2Msi` ?

Comment (zbud-msft) [reply sonic_data_client/db_client.go:82]: I don't see an issue with changing it to a public function for multiple packages to use. Will change to public.

**Review by qiluo-msft — APPROVED** [review]

**Comment by StormLiangMS** [comment]
@zbud-msft do you have ADO? test with 202305?

**Comment by zbud-msft** [comment]
> @zbud-msft do you have ADO? test with 202305?

@StormLiangMS ADO # is 24996556, Yes tested on 202305.

**Comment by yxieca** [comment]
@zbud-msft can you open an PR for 202205 branch? This chagne cannot be cherry-picked cleanly.

**Pull request merged**